crates/proof-of-sql/src/sql/parse/query_context_builder.rs

```diff
@@ -274,8 +274,8 @@ impl<'a> QueryContextBuilder<'a> {
 
 /// TODO: add docs
 pub(crate) fn type_check_binary_operation(
-    left_dtype: &ColumnType,
-    right_dtype: &ColumnType,
```

Comment:
NIT: Its not clear that these changes are related to the timestamp changes. This might be a better change in a separate PR.

Reply:
The changes to timestamp made `ColumnType` a smaller type, which made clippy request these changes.

Reply:
Ok, feel free to resolve this NIT then.